src/main/java/org/jenkinsci/plugins/workflow/support/pickles/TryRepeatedly.java

```diff
@@ -26,20 +26,36 @@
 
 import com.google.common.util.concurrent.AbstractFuture;
 import com.google.common.util.concurrent.ListenableFuture;
+import hudson.console.ModelHyperlinkNote;
+import hudson.model.TaskListener;
+import java.io.IOException;
 import jenkins.util.Timer;
 
 import java.util.concurrent.ScheduledFuture;
 import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
 import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import org.jenkinsci.plugins.workflow.flow.FlowExecutionOwner;
+import org.jenkinsci.plugins.workflow.pickles.Pickle;
 
 /**
  * {@link ListenableFuture} that promises a value that needs to be periodically tried.
- *
- * @author Kohsuke Kawaguchi
+ * Specialized for use from {@link Pickle#rehydrate(FlowExecutionOwner)}.
  */
 public abstract class TryRepeatedly<V> extends AbstractFuture<V> {
+
+    private static final Logger LOGGER = Logger.getLogger(TryRepeatedly.class.getName());
+
     private final int delay;
     private ScheduledFuture<?> next;
+    /** Number of {@link #tryLater} calls to run between logging attempts. */
+    private float backoff = 1;
+    /** Amount by which {@link #backoff} gets multiplied, so we do not flood the log with endless messages. */
+    private static final float BACKOFF_EXPONENT = 1.42f; // >√2
```

Comment:
:-)

The exponent here is pretty arbitrary, I just chose something which seemed to behave reasonably by watching it interactively.

`DurableTaskStep.Execution` uses a lower exponent, but there we want to actually grab process output sooner rather than later, whereas here the message is usually redundant, we just want to make it clear in the build log that something is still happening.